x-pack/test/functional/apps/apm/correlations/latency_correlations.ts

```diff
@@ -0,0 +1,137 @@
+        await PageObjects.common.navigateToApp('apm');
+        await testSubjects.existOrFail('apmMainContainer', {
+          timeout: 10000,
+        });
+
+        const apmMainContainerText = await testSubjects.getVisibleTextAll('apmMainContainer');
+        const apmMainContainerTextItems = apmMainContainerText[0].split('\n');
+        expect(apmMainContainerTextItems).to.contain('No services found');
+      });
+
+      it('sets the timePicker to return data', async () => {
+        await PageObjects.timePicker.timePickerExists();
+
+        const fromTime = 'Jul 29, 2019 @ 00:00:00.000';
+        const toTime = 'Jul 30, 2019 @ 00:00:00.000';
+        await PageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+
+        await retry.try(async () => {
+          const apmMainContainerText = await testSubjects.getVisibleTextAll('apmMainContainer');
+          const apmMainContainerTextItems = apmMainContainerText[0].split('\n');
+
+          expect(apmMainContainerTextItems).to.not.contain('No services found');
+
+          expect(apmMainContainerTextItems).to.contain('opbeans-go');
+          expect(apmMainContainerTextItems).to.contain('opbeans-node');
+          expect(apmMainContainerTextItems).to.contain('opbeans-ruby');
+          expect(apmMainContainerTextItems).to.contain('opbeans-python');
+          expect(apmMainContainerTextItems).to.contain('opbeans-dotnet');
+          expect(apmMainContainerTextItems).to.contain('opbeans-java');
+
+          expect(apmMainContainerTextItems).to.contain('development');
+
+          const items = await testSubjects.findAll('apmServiceListAppLink');
+          expect(items.length).to.be(6);
+        });
+      });
+
+      it(`navigates to the 'opbeans-go' service overview page`, async function () {
+        await find.clickByDisplayedLinkText(testData.serviceName);
+
+        await retry.try(async () => {
+          const apmMainTemplateHeaderServiceName = await testSubjects.getVisibleTextAll(
+            'apmMainTemplateHeaderServiceName'
+          );
+          expect(apmMainTemplateHeaderServiceName).to.contain('opbeans-go');
+        });
+      });
+
+      it('shows the correlations flyout', async function () {
+        await testSubjects.click('apmViewCorrelationsButton');
+
+        await retry.try(async () => {
+          await testSubjects.existOrFail('apmCorrelationsFlyout', {
+            timeout: 10000,
+          });
+
+          const apmCorrelationsFlyoutHeader = await testSubjects.getVisibleText(
+            'apmCorrelationsFlyoutHeader'
+          );
+
+          expect(apmCorrelationsFlyoutHeader).to.contain('Correlations BETA');
+        });
+      });
+
+      it('loads the correlation results', async function () {
+        await retry.try(async () => {
+          // Assert that the data fully loaded to 100%
+          const apmCorrelationsFlyoutHeader = await testSubjects.getVisibleText(
+            'apmCorrelationsLatencyCorrelationsProgressTitle'
```

Comment:
Nit: variable name doesn't really match the content. For the other element test variables I can see a pattern of naming the variable similar to the test subject, maybe that would fit here as well?

Reply:
Thanks for spotting that! Fixed in 2b2872c.